main.ts

```diff
@@ -44,20 +39,27 @@ export interface BeautitabPluginSettings {
 	inlineSearchProvider: SearchProvider;
 	showRecentFiles: boolean;
 	showQuote: boolean;
+	searchProviders: string[]; // Will be a list of plugin IDs, manually set by the user in data.json
 }
 
 const DEFAULT_SETTINGS: BeautitabPluginSettings = {
 	backgroundTheme: BackgroundTheme.SEASONS_AND_HOLIDAYS,
 	customBackground: "",
 	showTopLeftSearchButton: true,
-	topLeftSearchProvider: SearchProvider.SWITCHER,
+	topLeftSearchProvider: DEFAULT_SEARCH_PROVIDER,
 	showTime: true,
 	showGreeting: true,
 	greetingText: "Hello, Beautiful.",
 	showInlineSearch: true,
-	inlineSearchProvider: SearchProvider.SWITCHER,
+	inlineSearchProvider: DEFAULT_SEARCH_PROVIDER,
 	showRecentFiles: true,
 	showQuote: true,
+	searchProviders: [
```

Comment:
I don't think this needs to be a setting. It can still be a hard coded list of supported providers and we can add more later as they come up.

I want the plugin to be a simple as possible, usable by any user, and having manual editing of `data.json` does not align with that vision.

So still do what you're doing with the fuzzy search and etc, but move the list of providers to a constant. :)